commit b61b3e8b6355b155f0fbae00631b405060072708 (refs/changes/09/3209/1)
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2018-01-12T13:47:00-08:00 (1 year, 9 months ago)
    
    CNAPI-733 Add error for when cn-agent indicates docker cp'ing is not supported on a stopped container

diff --git a/lib/endpoints/vms.js b/lib/endpoints/vms.js
index 3ba821e..96aa5fd 100644
--- a/lib/endpoints/vms.js
+++ b/lib/endpoints/vms.js
@@ -1053,6 +1053,14 @@ VM.dockerCopy = function handlerVmDockerCopy(req, res, next) {
                 return;
             }
 
+            if (details.restCode ===
+                    'DockerCopyStoppedContainerNoPlatformSupport')
+            {
+                next(new
+                    errors.DockerCopyStoppedContainerNoPlatformSupportError());
+                return;
+            }
+
             next(new restify.InternalError(error.message));
             return;
         }
diff --git a/lib/errors.js b/lib/errors.js
index 0b83716..65a1779 100644
--- a/lib/errors.js
+++ b/lib/errors.js
@@ -210,6 +210,26 @@ FileNotFoundError.restCode = 'FileNotFound';
 FileNotFoundError.statusCode = 404;
 FileNotFoundError.description = 'could not find requested file';
 
+function DockerCopyStoppedContainerNoPlatformSupportError() {
+    _CnapiBaseError.call(this, {
+        restCode: this.constructor.restCode,
+        statusCode: this.constructor.statusCode,
+        message:
+            '"docker copy" on stopped container not supported on this joyent'
+            + ' platform version'});
+}
+
+util.inherits(
+    DockerCopyStoppedContainerNoPlatformSupportError, _CnapiBaseError);
+DockerCopyStoppedContainerNoPlatformSupportError.prototype.name =
+    'DockerCopyStoppedContainerNoPlatformSupportError';
+DockerCopyStoppedContainerNoPlatformSupportError.restCode =
+    'DockerCopyStoppedContainerNoPlatformSupport';
+DockerCopyStoppedContainerNoPlatformSupportError.statusCode = 409;
+DockerCopyStoppedContainerNoPlatformSupportError.description =
+    '"docker copy" on stopped container not supported on this joyent'
+    + ' platform version';
+
 
 function PathNotDirectoryError() {
     _CnapiBaseError.call(this, {
@@ -244,5 +264,7 @@ module.exports = {
     VmNotRunningError: VmNotRunningError,
     ServerNotRunningError: ServerNotRunningError,
     FileNotFoundError: FileNotFoundError,
-    PathNotDirectoryError: PathNotDirectoryError
+    PathNotDirectoryError: PathNotDirectoryError,
+    DockerCopyStoppedContainerNoPlatformSupportError:
+        DockerCopyStoppedContainerNoPlatformSupportError
 };
